### PR TITLE
[refactor](column) Simplify column functions

### PR DESCRIPTION
**Pull request by Gabriel39**
### What problem does this PR solve?

`insert_many_binary_data` has the same logics as `insert_many_strings`. This PR unify those 2 functions.

### Release note

None

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [x] No need to test or manual test. Explain why:
        - [x] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [x] Confirm the release note
- [x] Confirm test cases
- [x] Confirm document
- [x] Add branch pick label

### CODE REVIEW

**Comment by Gabriel39** [comment]
run buildall

File: be/src/vec/columns/column.h [hunk 1]
```diff
@@ -256,12 +256,6 @@ class IColumn : public COW<IColumn> {
                                "Method insert_many_dict_data is not supported for " + get_name());
     }
 
-    virtual void insert_many_binary_data(char* data_array, uint32_t* len_array,
```
Comment (yiguolei) [inline be/src/vec/columns/column.h:259]: why delete this method?

File: be/src/vec/columns/column.h [hunk 2]
```diff
@@ -256,12 +256,6 @@ class IColumn : public COW<IColumn> {
...
-                                         uint32_t* start_offset_array, size_t num) {
-        throw doris::Exception(ErrorCode::NOT_IMPLEMENTED_ERROR,
-                               "Method insert_many_binary_data is not supported for " + get_name());
-    }
-
     /// Insert binary data into column from a continuous buffer, the implementation maybe copy all binary data
     /// in one single time.
     virtual void insert_many_continuous_binary_data(const char* data, const uint32_t* offsets,
```
Comment (yiguolei) [inline be/src/vec/columns/column.h:261]: also modify this method?

**Review by yiguolei — DISMISSED** [review]

**Review by HappenLee — DISMISSED** [review]
LGTM

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

**Review by BiteTheDDDDt — APPROVED** [review]

**Review by yiguolei — APPROVED** [review]

**Pull request merged**